src/main/java/edu/ithaca/dragon/par/io/JsonDatastore.java

```diff
@@ -34,24 +37,69 @@ public JsonDatastore(String questionFilePath, String studentModelFilePath) throw
 
     @Override
     public List<StudentModel> loadStudentModels() throws IOException {
+         //check if filepath exists
         if(studentModelFilePath == null)
             throw new IOException("studentModelFilePath is null");
-        List<StudentModelRecord> studentModelRecords = JsonUtil.listFromJsonFile(studentModelFilePath, StudentModelRecord.class);
+
+        //make a list of studentModelRecords from every file in the filepath
+        List<StudentModelRecord> studentModelRecords = new ArrayList<>();
+        File studentModelFolder = new File(studentModelFilePath);
+        File[] listOfStudentModelFiles = studentModelFolder.listFiles();
+
+        for(File file : listOfStudentModelFiles){
+            if(file.isFile()){
+                String fullFileName = studentModelFilePath + "/" + file.getName();
+                StudentModelRecord newSMR = JsonUtil.fromJsonFile(fullFileName, StudentModelRecord.class);
+                studentModelRecords.add(newSMR);
+            }
+        }
+
+        //create studentModels from the studentModelRecords
         List<StudentModel> studentModels = new ArrayList<>();
         for(StudentModelRecord currSMR : studentModelRecords){
             studentModels.add(currSMR.buildStudentModel(questionPool));
         }
         return studentModels;
     }
 
+    @Override
+    public StudentModel loadStudentModel(String userId) throws IOException{
+        String fullFileName = studentModelFilePath + "/" + userId + ".json";
+
+        //check if file exists, return null if it doesn't
+        File checkFile = new File(fullFileName);
+        if(!checkFile.exists()){
+            return null;
+        }
+
+        StudentModelRecord SMR = JsonUtil.fromJsonFile(fullFileName, StudentModelRecord.class);
+        return SMR.buildStudentModel(questionPool);
+    }
+
     @Override
     public void saveStudentModels(Collection<StudentModel> studentModelsIn) throws IOException {
         if(studentModelFilePath == null)
             throw new IOException("studentModelFilePath is null");
-        List<StudentModelRecord> studentModelRecords = new ArrayList<>();
-        for(StudentModel currStudentModel : studentModelsIn)
-            studentModelRecords.add(new StudentModelRecord(currStudentModel));
-        JsonUtil.toJsonFile(studentModelFilePath, studentModelRecords);
+        for(StudentModel currStudentModel : studentModelsIn){
+            saveStudentModel(currStudentModel);
+        }
+    }
+
+    @Override
+    public void saveStudentModel(StudentModel studentModel) throws IOException{
+        if(studentModelFilePath == null)
+            throw new IOException("studentModelFilePath is null");
+        String fullFilePath = studentModelFilePath + "/" +  studentModel.getUserId() + ".json";
+        JsonUtil.toJsonFile(fullFilePath, new StudentModelRecord(studentModel));
     }
 
+    @Override
```

Comment:
Is this used? If not, we should leave it out...

Reply:
This was written for a test, but will probably never be used in the system